src/marqo/tensor_search/health.py

```diff
@@ -0,0 +1,57 @@
+from typing import Optional, Tuple
+import copy
+
+from marqo.config import Config
+from marqo import errors
+from marqo._httprequests import HttpRequests
+from marqo.tensor_search.enums import HealthStatuses
+
+
+def generate_heath_check_response(config: Config, index_name: Optional[str] = None) -> dict:
+    """Generate the health check response for check_heath(), check_index_health() APIs in tensor_search"""
+    marqo_status = get_marqo_status()
+    marqo_os_status = get_marqo_os_status(config, index_name=index_name)
+    marqo_status, marqo_os_status = aggregate_status(marqo_status, marqo_os_status)
```

Comment:
nit: to better reflect what is happening in the code you may want to rename `marqo_status`

```
aggregated_status, marqo_os_status = aggregate_status(...
```